src/hsm.cpp

```diff
@@ -104,6 +160,18 @@ openssl::SSL_EVP_PKEY_Ptr HsmEngine::generateKey(const RSASpec &spec,
                                                  const std::string &keyLabel,
                                                  const std::vector<uint8_t> &keyID)
 {
+    try {
+        // We need to make sure that we don't have 2 keys with the same ID.
+        // For that we need to pass empty keyLabel. Otherwise libp11 tries to find
+        // a key with exact keyLabel/keyID combination. This means that libp11 might
+        // not recognize that the key with the same ID is already there.
+        loadPrivateKey(keyID);
```

Comment:
So we don't expect public keys in the HSM without the corresponding private key?

Reply:
Pretty much yeah. The most common scenario I would say is that you have a keypair. I think having a private key without a corresponding public key also might be a valid scenario. But having a public key on HSM without the corresponding private doesn't make sense to me.

Reply:
at least if you have other mechanisms to protect the public key like secure boot + dm-verity. :)

So the question might be, if there will be ECUs using the keystore without having secure boot + dm-verity.

@garw: Any strong opinion on this?

I would be also fine with just going for the privat key.